library/std/src/sys/unix/fs.rs

```diff
@@ -1140,14 +1140,14 @@ pub fn copy(from: &Path, to: &Path) -> io::Result<u64> {
     }
 
     let (mut reader, reader_metadata) = open_from(from)?;
-    let len = reader_metadata.len();
+    let max_len = u64::MAX;
     let (mut writer, _) = open_to_and_set_permissions(to, reader_metadata)?;
 
     let has_copy_file_range = HAS_COPY_FILE_RANGE.load(Ordering::Relaxed);
     let mut written = 0u64;
-    while written < len {
+    while written < max_len {
         let copy_result = if has_copy_file_range {
-            let bytes_to_copy = cmp::min(len - written, usize::MAX as u64) as usize;
+            let bytes_to_copy = cmp::min(max_len - written, usize::MAX as u64) as usize;
```

Comment:
Is it really necessary to keep track of `max_len - written` accurately at this point?  It's probably okay just to pass a large constant size on every call.

Reply:
It avoids overflowing an u64. Plus I want to reuse the code in #75272 which will require exact max sizes